### PR TITLE
feat!: update to uno 5

### PR DESCRIPTION
**Pull request by juliecantin**
GitHub Issue: #

## Proposed Changes

 - [ ] Bug fix
 - [x] Feature
 - [ ] Code style update (formatting)
 - [ ] Refactoring (no functional changes, no api changes)
 - [ ] Build or CI related changes
 - [ ] Documentation content changes
 - [ ] Other, please describe:

## Description

Update to Uno 5

## Impact on version

- [x] **Major**
  - The template structure was changed.
- [ ] **Minor**
  - New functionalities were added.
- [ ] **Patch**
  - A bug in behavior was fixed.
  - Documentation was changed.

### Always applicable
No matter your changes, these checks always apply.
- [x] Your conventional commits are aligned with the **Impact on version** section.
- [ ] Updated [CHANGELOG.md](../CHANGELOG.md).
  - Use the latest Major.Minor.X header if you do a **Patch** change.
  - Create a new Major.Minor.X header if you do a **Minor** or **Major** change.
  - If you create a new header, it aligns with the **Impact on version** section and matches what is generated in the build pipeline.
- [x] Documentation files were updated according with the changes.
  - Update `README.md` and `TemplateConfig.md` if you made changes to **templating**.
  - Update `AzurePipelines.md` and `APP_README.md` if you made changes to **pipelines**.
  - Update `Diagnostics.md` if you made changes to **diagnostic tools**.
  - Update `Architecture.md` and its diagrams if you made **architecture decisions** or if you introduced new **recipes**.
  - ...and so forth: Make sure you update the documentation files associated to the recipes you changed. Review the topics by looking at the content of the `doc/` folder.
- [x] Images about the template are referenced from the [wiki](https://github.com/nventive/UnoApplicationTemplate/wiki/Images) and added as images in this git.
- [x] TODO comments are hints for next steps for users of the template and not planned work.

### Contextual
Based on your changes these checks may not apply.
- [ ] Automated tests for the changes have been added/updated.
- [ ] Tested on all relevant platforms

### CODE REVIEW

File: src/app/ApplicationTemplate.Shared.Views/App.xaml.cs [hunk 1]
```diff
@@ -43,6 +44,10 @@ public App()
 	protected override void OnLaunched(LaunchActivatedEventArgs args)
 	{
 		InitializeAndStart();
+
+#if DEBUG
```
Comment (Soap-141) [inline src/app/ApplicationTemplate.Shared.Views/App.xaml.cs:48]: Why not have this inside InitializeAndStart() ?

File: src/app/ApplicationTemplate.Mobile/ApplicationTemplate.Mobile.csproj [hunk 2]
```diff
@@ -17,25 +17,25 @@
...
-		<PackageReference Include="Uno.WinUI.RemoteControl" Version="4.8.24" Condition="'$(Configuration)'=='Debug'" />
-		<PackageReference Include="Uno.WinUI.Lottie" Version="4.8.24" />
-		<PackageReference Include="Uno.UI.Adapter.Microsoft.Extensions.Logging" Version="4.8.24" />
+		<PackageReference Include="Uno.CommunityToolkit.WinUI" Version="7.1.200" />
+		<PackageReference Include="Uno.Material.WinUI" Version="4.0.6" />
+		<PackageReference Include="Uno.Microsoft.Xaml.Behaviors.Interactivity.WinUI" Version="2.4.2" />
+		<PackageReference Include="Uno.Microsoft.Xaml.Behaviors.WinUI.Managed" Version="2.4.2" />
+		<PackageReference Include="Uno.Toolkit.WinUI.Material" Version="5.0.17" />
+		<PackageReference Include="Uno.WinUI" Version="5.0.64" />
+		<PackageReference Include="Uno.WinUI.DevServer" Version="5.0.19" Condition="'$(Configuration)'=='Debug'" />
```
Comment (carlh98) [inline src/app/ApplicationTemplate.Mobile/ApplicationTemplate.Mobile.csproj:36]: Update to 5.0.64 for the following
Uno.WinUI.DevServer
Uno.WinUI.Lottie
Uno.UI.Adapter.Microsoft.Extensions.Logging

File: src/app/ApplicationTemplate.Access/ApplicationTemplate.Access.csproj [hunk 3]
```diff
@@ -20,13 +20,13 @@
 		<PackageReference Include="Microsoft.Extensions.Http" Version="3.1.0" />
 		<PackageReference Include="Microsoft.Extensions.Logging.Abstractions" Version="7.0.1" />
 		<PackageReference Include="Microsoft.Extensions.Options.ConfigurationExtensions" Version="3.1.0" />
-		<PackageReference Include="Reactive.Annex" Version="0.6.0-dev.50" />
+		<PackageReference Include="Reactive.Annex" Version="2.0.0-feature.Uno5.2" />
```
Comment (jeanplevesque) [inline src/app/ApplicationTemplate.Access/ApplicationTemplate.Access.csproj:23]: I know this is still draft, but the plan is to update to stable versions (non `-feature`) before merging this PR, right?

Comment (Soap-141) [reply src/app/ApplicationTemplate.Access/ApplicationTemplate.Access.csproj:23]: I feel like It should.

Comment (Soap-141) [reply src/app/ApplicationTemplate.Access/ApplicationTemplate.Access.csproj:23]: Probably just to test

Comment (carlh98) [reply src/app/ApplicationTemplate.Access/ApplicationTemplate.Access.csproj:23]: yes

Comment (carlh98) [reply src/app/ApplicationTemplate.Access/ApplicationTemplate.Access.csproj:23]: @jeanplevesque @Soap-141 does that sound good to you guys?

Comment (jeanplevesque) [reply src/app/ApplicationTemplate.Access/ApplicationTemplate.Access.csproj:23]: Yes that works now that the target branch was changed :)
image

File: CHANGELOG.md [hunk 4]
```diff
@@ -5,14 +5,21 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/)
 
 Prefix your items with `(Template)` if the change is about the template and not the resulting application.
 
+## 3.0.X
+- Updated to Uno 5
```
Comment (carlh98) [inline CHANGELOG.md:9]: add a dot a the end

File: build/templates/dotnet-install.yml [hunk 5]
```diff
@@ -1,15 +1,19 @@
...
+    packageType: 'sdk'
+    version: ${{ parameters.DotNetVersion }}
 
-  - powershell: dotnet workload install maui --source https://aka.ms/dotnet6/nuget/index.json --source https://api.nuget.org/v3/index.json
-    condition: ${{ parameters.installMobileWorkload }}
-    displayName: Install .NET MAUI Workload
+- powershell: |
+    & dotnet tool update --global uno.check --version ${{ parameters.UnoCheck_Version }} --add-source https://api.nuget.org/v3/index.json
+    & uno-check -v --ci --non-interactive --fix --skip xcode --skip gtk3 --skip vswin --skip vsmac --skip androidsdk --skip androidemulator --manifest ${{ parameters.UnoCheck_Manifest }}
+  displayName: Install .NET Workloads | Uno-check
```
Comment (Soap-141) [inline build/templates/dotnet-install.yml:16]: Display name could be Install .NET Workloads | Uno Check instead :)

**Review by jeanplevesque — APPROVED** [review]

**Review by Soap-141 — APPROVED** [review]

**Pull request merged**